src/modules/videoconference/VideoConferenceModule/Toolbar.cpp

```diff
@@ -154,18 +154,27 @@ LRESULT Toolbar::WindowProcessMessages(HWND hwnd, UINT msg, WPARAM wparam, LPARA
         static bool previousShow = false;
         bool show = false;
 
-        if (toolbar->cameraInUse)
+        if (toolbar->ToolbarHide == L"Never")
         {
-            show = toolbar->HideToolbarWhenUnmuted ? toolbar->microphoneMuted || toolbar->cameraMuted : true;
+            show = true;
         }
-        else if (toolbar->previouscameraInUse)
+        else if (toolbar->ToolbarHide == L"When both camera and microphone are muted")
         {
-            VideoConferenceModule::unmuteAll();
+            if(!toolbar->previouscameraInUse && toolbar->cameraInUse && !toolbar->moduleSettingsUpdateScheduled)
+            {
+                VideoConferenceModule::muteAll();
```

Comment:
I understand the intention, but we shouldn't mute everything, if a UI-related setting has a certain value. Maybe let's do a separate setting (could be also done in a separate PR).

Reply:
Something like giving user the option to click the "Mute the camera automatically when it starts" checkbox if the "Hide when mute everything is disabled" option is selected?

Reply:
Yes, but let's do it in a separate PR please.